projects/cdk/directives/for-async/for-async.directive.ts

```diff
@@ -0,0 +1,47 @@
+import {
+    Directive,
+    Inject,
+    Input,
+    NgZone,
+    OnDestroy,
+    TemplateRef,
+    ViewContainerRef,
+} from '@angular/core';
+import {TuiDestroyService} from '@taiga-ui/cdk/services';
+import {from, of} from 'rxjs';
+import {concatMap, delay, takeUntil} from 'rxjs/operators';
+
+@Directive({
+    selector: '[tuiForAsync][tuiForAsyncOf]',
+    providers: [TuiDestroyService],
+})
+export class TuiForAsyncDirective<T> implements OnDestroy {
+    @Input() tuiForAsyncTimeout = 10;
+
+    constructor(
+        @Inject(ViewContainerRef) private readonly view: ViewContainerRef,
+        @Inject(TemplateRef) private readonly template: TemplateRef<unknown>,
+        @Inject(NgZone) private readonly ngZone: NgZone,
+        @Inject(TuiDestroyService) private readonly destroy$: TuiDestroyService,
+    ) {}
+
+    @Input()
```

Comment:
What if the input changes? Shouldn't we clear old view?

Reply:
fixed